src/code/mixins/app-view.coffee

```diff
@@ -20,27 +20,25 @@ module.exports =
           deleteFunction()
     else
       $(window).off 'keydown'
+      
+  addToPalette: (node) ->
+    if node?.image.match /^(https?|data):/
+      # make sure this is a new image
+      if not _.find @state.protoNodes, {image: node.image}
```

Comment:
Do we need a Palette model here so that the view doesn't have to interrogate the data so hard?  This Palette model could have an "addToPalette" method which could do these checks and handle ordering &etc.